clang/test/CodeGen/X86/avx10_2_512bf16-builtins.c

```diff
@@ -0,0 +1,1054 @@
+  return _mm512_cmp_pbh_mask(a, b, _CMP_UNORD_S);
+}
+
+__mmask32 test_mm512_cmp_pbh_mask_neq_us(__m512bh a, __m512bh b) {
+  // CHECK-LABEL: test_mm512_cmp_pbh_mask_neq_us
+  // CHECK: fcmp une <32 x bfloat> %{{.*}}, %{{.*}}
+  return _mm512_cmp_pbh_mask(a, b, _CMP_NEQ_US);
+}
+
+__mmask32 test_mm512_cmp_pbh_mask_nlt_uq(__m512bh a, __m512bh b) {
+  // CHECK-LABEL: test_mm512_cmp_pbh_mask_nlt_uq
+  // CHECK: fcmp uge <32 x bfloat> %{{.*}}, %{{.*}}
+  return _mm512_cmp_pbh_mask(a, b, _CMP_NLT_UQ);
+}
+
+__mmask32 test_mm512_cmp_pbh_mask_nle_uq(__m512bh a, __m512bh b) {
+  // CHECK-LABEL: test_mm512_cmp_pbh_mask_nle_uq
+  // CHECK: fcmp ugt <32 x bfloat> %{{.*}}, %{{.*}}
+  return _mm512_cmp_pbh_mask(a, b, _CMP_NLE_UQ);
+}
+
+__mmask32 test_mm512_cmp_pbh_mask_ord_s(__m512bh a, __m512bh b) {
+  // CHECK-LABEL: test_mm512_cmp_pbh_mask_ord_s
+  // CHECK: fcmp ord <32 x bfloat> %{{.*}}, %{{.*}}
+  return _mm512_cmp_pbh_mask(a, b, _CMP_ORD_S);
+}
+
+__mmask32 test_mm512_cmp_pbh_mask_eq_us(__m512bh a, __m512bh b) {
+  // CHECK-LABEL: test_mm512_cmp_pbh_mask_eq_us
+  // CHECK: fcmp ueq <32 x bfloat> %{{.*}}, %{{.*}}
+  return _mm512_cmp_pbh_mask(a, b, _CMP_EQ_US);
+}
+
+__mmask32 test_mm512_cmp_pbh_mask_nge_uq(__m512bh a, __m512bh b) {
+  // CHECK-LABEL: test_mm512_cmp_pbh_mask_nge_uq
+  // CHECK: fcmp ult <32 x bfloat> %{{.*}}, %{{.*}}
+  return _mm512_cmp_pbh_mask(a, b, _CMP_NGE_UQ);
+}
+
+__mmask32 test_mm512_cmp_pbh_mask_ngt_uq(__m512bh a, __m512bh b) {
+  // CHECK-LABEL: test_mm512_cmp_pbh_mask_ngt_uq
+  // CHECK: fcmp ule <32 x bfloat> %{{.*}}, %{{.*}}
+  return _mm512_cmp_pbh_mask(a, b, _CMP_NGT_UQ);
+}
+
+__mmask32 test_mm512_cmp_pbh_mask_false_os(__m512bh a, __m512bh b) {
+  // CHECK-LABEL: test_mm512_cmp_pbh_mask_false_os
+  // CHECK: fcmp false <32 x bfloat> %{{.*}}, %{{.*}}
+  return _mm512_cmp_pbh_mask(a, b, _CMP_FALSE_OS);
+}
+
+__mmask32 test_mm512_cmp_pbh_mask_neq_os(__m512bh a, __m512bh b) {
+  // CHECK-LABEL: test_mm512_cmp_pbh_mask_neq_os
+  // CHECK: fcmp one <32 x bfloat> %{{.*}}, %{{.*}}
+  return _mm512_cmp_pbh_mask(a, b, _CMP_NEQ_OS);
+}
+
+__mmask32 test_mm512_cmp_pbh_mask_ge_oq(__m512bh a, __m512bh b) {
+  // CHECK-LABEL: test_mm512_cmp_pbh_mask_ge_oq
+  // CHECK: fcmp oge <32 x bfloat> %{{.*}}, %{{.*}}
+  return _mm512_cmp_pbh_mask(a, b, _CMP_GE_OQ);
+}
+
+__mmask32 test_mm512_cmp_pbh_mask_gt_oq(__m512bh a, __m512bh b) {
+  // CHECK-LABEL: test_mm512_cmp_pbh_mask_gt_oq
+  // CHECK: fcmp ogt <32 x bfloat> %{{.*}}, %{{.*}}
+  return _mm512_cmp_pbh_mask(a, b, _CMP_GT_OQ);
+}
+
+__mmask32 test_mm512_cmp_pbh_mask_true_us(__m512bh a, __m512bh b) {
+  // CHECK-LABEL: test_mm512_cmp_pbh_mask_true_us
+  // CHECK: fcmp true <32 x bfloat> %{{.*}}, %{{.*}}
+  return _mm512_cmp_pbh_mask(a, b, _CMP_TRUE_US);
+}
+
+__mmask32 test_mm512_mask_cmp_pbh_mask_eq_oq(__mmask32 m, __m512bh a, __m512bh b) {
+  // CHECK-LABEL: @test_mm512_mask_cmp_pbh_mask_eq_oq
+  // CHECK: fcmp oeq <32 x bfloat> %{{.*}}, %{{.*}}
```

Comment:
Check select for mask version?

Reply:
There's only a following and. I can add a check for that. clang emit an extra and for _mask_cmp_ intrinsics, which is a gap with gcc: https://gcc.godbolt.org/z/5n63rYEqE

Reply:
[7957436](https://github.com/llvm/llvm-project/pull/101603/commits/7957436ac07d1902f3f510de756577c00353dd02)